src/RealFourierArray.jl

```diff
@@ -12,46 +13,41 @@ import Base.eltype, Base.size, Base.length
 
 @enum RFA_state RFA_only_real RFA_only_fourier RFA_both
 
-# TODO make this concrete by parametrizing on the type of the arrays
 """
-A structure to facilitate manipulations of an array of type T in both real
+A structure to facilitate manipulations of an array of real-space type T, in both real
 and fourier space. Create with `from_real` or `from_fourier`, and access
 with `A.real` and `A.fourier`.
 """
-mutable struct RealFourierArray{Treal <: Real, T <: Union{Treal, Complex{Treal}}}
-# Treal is the underlying real type
-# T is the type of the array in real space
-    basis::PlaneWaveBasis{Treal}
+mutable struct RealFourierArray{T <: Real}
+    basis::PlaneWaveBasis{T}
     _real::AbstractArray{T, 3}
-    _fourier::AbstractArray{Complex{Treal}, 3}
+    _fourier::AbstractArray{Complex{T}, 3}
     _state::RFA_state
 end
 # Type of the real part
-Base.eltype(::RealFourierArray{Treal, T}) where {Treal, T} = T
+Base.eltype(::RealFourierArray{T}) where {T} = T
 Base.size(array::RealFourierArray) = size(array._real)
 Base.size(array::RealFourierArray, i) = size(array._real, i)
 Base.length(array::RealFourierArray) = length(array._real)
 
 # Constructors
-RealFourierArray(basis, real, fourier) = RealFourierArray(basis, real, fourier, 2)
-function RealFourierArray(basis::PlaneWaveBasis{T}; iscomplex=false) where {T}
-    from_real(basis, zeros(iscomplex ? complex(T) : T, basis.fft_size...))
+RealFourierArray(basis, real, fourier) = RealFourierArray(basis, real, fourier, RFA_both)
```

Comment:
I think you need a `{T}` here.

Reply:
? why?

Reply:
How else would it resolve the type of the RealFourierArray. Maybe I'm wrong, but I have in mind in such cases you have to help Julia a little.

Reply:
well from the types of its arguments. There's a single match so it'll just take that. Anyway it didn't seem to complain...

Reply:
true. So it seems it works.